### PR TITLE
[Gallery] Preserve demo state, simplify demo width calculations on desktop

### PR DESCRIPTION
**Pull request by guidezpl**
TLDR: Fixes a bugs on both desktop and mobile which caused demo state to be reset. 
This was caused by rendering different trees; on mobile: based on whether the options/info panels were open; on desktop, based on whether the demo was in fullscreen. Additionally on desktop, it seemed that conditionally rendering the sized box (or container) preceding the demoContent caused the demoContent widget to be fully rebuilt. Instead, I always specify its width.

This PR also removes width calculations, instead specifying a max width for info and options panes. This incidentally causes the demoContent to remain the same width regardless of the chosen section on desktop. The demoContent grows with the screen size, as before. The section on the left grows until the specified max width.

Screen Shot 2020-01-28 at 11 57 23 AM
Screen Shot 2020-01-28 at 11 57 27 AM
Screen Shot 2020-01-28 at 11 57 29 AM
Screen Shot 2020-01-28 at 11 57 39 AM

Closes https://github.com/material-components/material-components-flutter-gallery/issues/312

### CODE REVIEW

File: gallery/gallery/lib/pages/demo.dart [hunk 1]
```diff
@@ -314,46 +316,15 @@ class _DemoPageState extends State<DemoPage> with TickerProviderStateMixin {
...
-                  child: demoContent,
-                ),
-              ],
-            );
+      final isFullScreen = _state == _DemoState.fullscreen;
+      final Widget sectionAndDemo = Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          if (!isFullScreen) Expanded(child: section),
+          SizedBox(width: !isFullScreen ? 48.0 : 0),
```
Comment (rami-a) [inline gallery/gallery/lib/pages/demo.dart]: I think this might be easier to read as:

```
if (!isFullScreen) ...[
  Expanded(child: section),
  SizedBox(),
]
Expanded(child: demoContent),
```

File: gallery/gallery/lib/pages/demo.dart [hunk 2]
```diff
@@ -371,20 +342,23 @@ class _DemoPageState extends State<DemoPage> with TickerProviderStateMixin {
...
-          label: MaterialLocalizations.of(context).modalBarrierDismissLabel,
-          child: GestureDetector(
-            behavior: HitTestBehavior.opaque,
-            onTap: () {
+      demoContent = Semantics(
+        label: MaterialLocalizations.of(context).modalBarrierDismissLabel,
+        child: GestureDetector(
+          behavior: HitTestBehavior.opaque,
+          onTap: () {
+            if (_state != _DemoState.normal) {
```
Comment (rami-a) [inline gallery/gallery/lib/pages/demo.dart]: Since the gestureDetector is added always now, does this interfere with interacting with the demo?

File: gallery/gallery/lib/pages/demo.dart [hunk 3]
```diff
@@ -506,49 +482,52 @@ class _DemoSectionOptions extends StatelessWidget {
...
-              top: 12,
-              end: 24,
-            ),
-            child: Text(
-              GalleryLocalizations.of(context).demoOptionsTooltip,
-              style: textTheme.display1.apply(
-                color: colorScheme.onSurface,
-                fontSizeDelta:
-                    isDisplayDesktop(context) ? desktopDisplay1FontDelta : 0,
+    return Align(
```
Comment (rami-a) [inline gallery/gallery/lib/pages/demo.dart]: What is the `Align` for?

File: gallery/gallery/lib/pages/demo.dart (hunk 1 above)
Comment (guidezpl) [reply gallery/gallery/lib/pages/demo.dart]: This is what I was saying causes the demoContent to be rebuilt from scratch. I could write it as 

```dart
if (!isFullScreen) ...[
  Expanded(child: section),
  SizedBox(),
]
Expanded(child: demoContent),
```
and it works

Comment (rami-a) [reply gallery/gallery/lib/pages/demo.dart]: Very strange...ok I think I prefer what you originally had instead of the 2nd sizedbox

File: gallery/gallery/lib/pages/demo.dart (hunk 2 above)
Comment (guidezpl) [reply gallery/gallery/lib/pages/demo.dart]: I've tested on chrome, iOS and macOS, and saw no adverse effects. I'll remove behavior: HitTestBehavior.opaque because it's not doing anything here and might be confusing

File: gallery/gallery/lib/pages/demo.dart (hunk 3 above)
Comment (guidezpl) [reply gallery/gallery/lib/pages/demo.dart]: A Container within an Expanded will fill the space, even if you set width/height constraints, unless you use an Align.
